test/function/timestamp_functions_test.cpp

```diff
@@ -168,5 +168,141 @@ TEST_F(TimestampFunctionsTests, DateTruncYearTest) {
   DateTruncTestHelper(DatePartType::YEAR, date, expected);
 }
 
+/**
+ * Helper method for TimestampFunctions::DatePart
+ * It packages up the inputs into the right format
+ * and checks whether we get the correct result.
+ */
+void DatePartTestHelper(DatePartType part, std::string &date,
+                         std::string &expected) {
+  // The first argument is an VarcharValue of the DatePartType
+  // The second argument is a TimestampValue of the date
+  std::string string_date_part = DatePartTypeToString(part);
+  const char *char_date_part = string_date_part.c_str();
+  uint64_t int_date =
+      type::ValueFactory::CastAsTimestamp(
+          type::ValueFactory::GetVarcharValue(date)).GetAs<uint64_t>();
+
+  // The expected return value
+  double double_expected =
+      type::ValueFactory::CastAsDecimal(
+          type::ValueFactory::GetVarcharValue(expected)).GetAs<double>();
+
+  // Invoke the DatePart method and get back the result
+  auto result =
+      function::TimestampFunctions::DatePart(char_date_part, int_date);
+
+  LOG_INFO("COMPARED: %s, %f = %f \n",
+        expected.c_str(), double_expected,
+        result);
+
+  // Check that result is *not* null
+  EXPECT_NE(result, type::PELOTON_DECIMAL_NULL);
```

Comment:
For these expectations, the expected value is the first argument and the actual value is the second argument. Can you change these `EXPECT_...` assertions appropriately?